### DISCUSSION TITLE
`CustomTick` component in `YAxis` not using all available space and not showing icon

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [recharts/recharts#3563] [Q&A]
Hey there 👋,

I have implemented a `CustomTick` component for the `YAxis`, which should display a text label and an icon if the label matches a certain value. However, the `CustomTick` component is not using all the available space for the text label and the icon is not showing, but instead of the icon showing, it is rendering `[object, object]`.

Current output example:

<img width="521" alt="Tick" src="https://user-images.githubusercontent.com/55815265/235664912-ebb92b5f-f899-4390-95d7-c8211a504ad9.png">

My `CustomTick` comp:

```jsx
function CustomTick(props) {
  const {
    payload: { value }
  } = props;
  const isWithIcon = NAME_WITH_ICON.includes(value);

  return (
    <Text {...props} fill={isWithIcon ? "green" : "red"}>
      {isWithIcon && <GoGlobe />}
      {value}
    </Text>
  );
}
```

What could be causing the `CustomTick` component to not use all the available space for the text and not show the icon?

Here is a code example: https://codesandbox.io/s/recharts-custom-tick-with-a-icon-upiev7

Any help would be appreciated. Thank you!

**Comment by aamazing-coder** [comment]
Since you are using a margin to give that space to the left of the graphic, you can use a fixed width for the text (`width={160}`):

```tsx
function CustomTick(props) {
   const {
     payload: { value }
   } = props;
   const isWithIcon = NAME_WITH_ICON.includes(value);

   return(
     <Text {...props} width={160} fill={isWithIcon ? "green" : "red"}>
       {isWithIcon && <GoGlobe />}
       {value}
     </Text>
   );
}
```

About the icon, I don't know how to help you. But it's weird because recharts uses SVG, and react-icons are SVGs too, so in theory it should work.

**Reply by 201flaviosilva** [reply]
Ohh, nice, that fixed the size problem, thanks a lot.

Yeah, I don’t understand, it’s really weird 😅, but thanks any way :)
